dashboard/src/main/java/com/google/cloud/tools/opensource/dashboard/DashboardMain.java

```diff
@@ -60,71 +62,71 @@ public static void main(String[] args)
 
   public static Path generate() throws IOException, TemplateException, ArtifactDescriptorException {
     Configuration configuration = configureFreemarker();
-    
+
     Path relativePath = Paths.get("target", "dashboard");
     Path output = Files.createDirectories(relativePath);
-    
+
     DefaultArtifact bom =
         new DefaultArtifact("com.google.cloud:cloud-oss-bom:pom:0.62.0-SNAPSHOT");
     List<Artifact> managedDependencies = RepositoryUtility.readBom(bom);
-    
+
     List<ArtifactResults> table = generateReports(configuration, output, managedDependencies);
     generateDashboard(configuration, output, table);
-    
+
     return output;
   }
 
-  private static Configuration configureFreemarker() {
+  @VisibleForTesting
+  static Configuration configureFreemarker() {
     Configuration configuration = new Configuration(new Version("2.3.28"));
     configuration.setDefaultEncoding("UTF-8");
     configuration.setClassForTemplateLoading(DashboardMain.class, "/");
     return configuration;
   }
 
-  private static List<ArtifactResults> generateReports(Configuration configuration, Path output,
+  @VisibleForTesting
+  static List<ArtifactResults> generateReports( Configuration configuration, Path output,
       List<Artifact> artifacts) {
-    
+
     List<ArtifactResults> table = new ArrayList<>();
 
-    for (Artifact artifact : artifacts ) {
+    for (Artifact artifact : artifacts) {
       try {
         ArtifactResults results = generateReport(configuration, output, artifact);
         table.add(results);
-      } catch (DependencyCollectionException | DependencyResolutionException | IOException
-          | TemplateException ex) {
-        // TODO the dashboard should somehow show that it failed to generate this report;
-        // not just silently drop it from the index
-        System.err.println("Error generating report for " + artifact);
-        System.err.println(ex.getMessage());
+      } catch (RepositoryException | IOException | TemplateException ex) {
```

Comment:
TemplaetException needs to be separated into its own catch block with different behavior

Reply:
As talked:

- IOException -> show in dashboard
- TemplateException -> RuntimeException